controllers/ibu_controller.go

```diff
@@ -194,26 +204,40 @@ func (r *ImageBasedUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.Re
 					return
 				}
 				if !isValid {
-					err = utils.UpdateIBUStatus(ctx, r.Client, ibu)
+					ibu.Status.ValidNextStages = getValidNextStageList(ibu, isAfterPivot)
+					if updateErr := utils.UpdateIBUStatus(ctx, r.Client, ibu); updateErr != nil {
+						r.Log.Error(updateErr, "failed to update IBU CR status")
+					}
 					return
 				}
 			}
-			nextReconcile = requeueImmediately()
-		}
-	} else {
-		inProgressStage := utils.GetInProgressStage(ibu)
-		if inProgressStage != "" {
-			nextReconcile, err = r.handleStage(ctx, ibu, inProgressStage)
-			if err != nil {
-				_ = utils.UpdateIBUStatus(ctx, r.Client, ibu)
-				return
+			ibu.Status.ValidNextStages = getValidNextStageList(ibu, isAfterPivot)
+			// Update status
+			if updateErr := utils.UpdateIBUStatus(ctx, r.Client, ibu); updateErr != nil {
```

Comment:
Not sure if continuing reconciliation is the best option when there is an update err

Reply:
I was originally just ignoring the interim status update failures, but figured it would be better to log them at least. I think all the DB interactions should be reviewed/revised, as the current Update potentially masks issues.